src/models.ts

```diff
@@ -1,9 +1,345 @@
-import * as dashboard from './dashboard';
-import * as report from './report';
-import {IError} from './util';
-
-export {
-    report,
-    dashboard,
-    IError,
-};
+declare var require: Function;
+
+/* tslint:disable:no-var-requires */
+export const advancedFilterSchema = require('./schemas/advancedFilter.json');
+export const filterSchema = require('./schemas/filter.json');
+export const loadSchema = require('./schemas/reportLoadConfiguration.json');
+export const dashboardLoadSchema = require('./schemas/dashboardLoadConfiguration.json');
+export const pageSchema = require('./schemas/page.json');
+export const settingsSchema = require('./schemas/settings.json');
+export const basicFilterSchema = require('./schemas/basicFilter.json');
+/* tslint:enable:no-var-requires */
+
+import * as jsen from 'jsen';
+
+interface IValidationError {
+  path: string;
+  keyword: string;
+  message: string;
+}
+
+export interface IError {
+  message: string;
+}
+
+function normalizeError(error: IValidationError): IError {
+  let message = error.message;
+
+  if (!message) {
+    message = `${error.path} is invalid. Not meeting ${error.keyword} constraint`;
+  }
+
+  return {
+    message
+  };
+}
+
+/**
+ * Takes in schema and returns function which can be used to validate the schema with better semantics around exposing errors
+ */
+function validate(schema: any, options?: any) {
+  return (x: any): IError[] => {
+    const validate = jsen(schema, options);
+    const isValid = validate(x);
+
+    if (isValid) {
+      return undefined;
+    }
+    else {
+      return validate.errors
+        .map(normalizeError);
+    }
+  };
+}
+
+export interface ISettings {
+  filterPaneEnabled?: boolean;
+  navContentPaneEnabled?: boolean;
+}
+
+export const validateSettings = validate(settingsSchema, {
+  schemas: {
+    basicFilter: basicFilterSchema,
+    advancedFilter: advancedFilterSchema
+  }
+});
+
+export interface ILoadConfiguration {
```

Comment:
We could change ILoadConfiguration to IReportLoadConfiguration as you suggested, makes sense